src/main.rs

```diff
@@ -15,9 +19,21 @@ use crate::{
     input::add_input,
 };
 
+fn print_completions<G: Generator>(gen: G, cmd: &mut Command) {
+    generate(gen, cmd, cmd.get_name().to_string(), &mut io::stdout());
+}
+
 fn main() -> Result<(), Box<dyn Error>> {
+let matches = build_cli().get_matches();
     let mut cli = Cli::from_args();
 
+    if let Some(generator) = matches.get_one::<Shell>("gencompletions") {
+        let mut cmd = build_cli();
```

Comment:
You should be able to do `Cli::command()` here instead of calling `build_cli()` 😄